File: rental-backend/src/main/java/com/realestate/rental/dto/NotificationDto.java
package com.realestate.rental.dto;

import java.time.LocalDateTime;
import java.util.UUID;

public record NotificationDto(
        UUID id,
        String title,
        String message,
        String type,
        String relatedEntityType,
        UUID relatedEntityId,
        Boolean isRead,
        LocalDateTime createdAt
) {}

File: rental-app-flutter/lib/features/notifications/models/notification_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'notification_dto.freezed.dart';
part 'notification_dto.g.dart';

@freezed
abstract class NotificationDto with _$NotificationDto {
  const factory NotificationDto({
    required String id,
    required String title,
    required String message,
    required String type,
    String? relatedEntityType,
    String? relatedEntityId,
    required bool isRead,
    required String createdAt,
  }) = _NotificationDto;

  factory NotificationDto.fromJson(Map<String, dynamic> json) => _$NotificationDtoFromJson(json);
}
